src/Framework/Framework/Compilation/Parser/Binding/Parser/BindingParser.cs

```diff
@@ -47,6 +47,75 @@ public BindingParserNode ReadDirectiveValue()
             return first;
         }
 
+        public BindingParserNode ReadArrayInitializerValue()
```

Comment:
It seems that you gave this expression very low precedence (bellow OrElse), while it should be on par with other atomic expressions (constants, parenthesized exprs). For example `[1, 2, 3].Where(x => x % 2 == 0)` won't work this way (I think)

Reply:
The question is: Should it work. I intentionally wanted to prevent "creative" uses of the array initializer ;)

Reply:
Well, `([1, 2, 3]).Where(...)` will still work (AFAIK), so this isn't actually preventing anything